### ISSUE TITLE
Document Upload fails on instances migrated on ITZBund and 1.59.1

### ISSUE THREAD

**Issue by maxiheyner** [hzi-braunschweig/SORMAS-Project#5452]
### Bug Description
Two GAs reported failing document uploads for cases and contacts after being migrated on ITZBund servers and on version 1.59.1 last week. We are not able to reproduce these failing uploads on our Testsystems with version 1.59.1 since those are not ITZBund hosted. 

### Steps to Reproduce

1. use a Sormas instance hosted by ITZ Bund with Version 1.59.1
2. got to a case/contact
3. start an upload (each format and sizes up to 10MB were affected)
4. the upload seemingly starts but gets interrupted then

### Expected Behavior
All uploads should work in an appropriate timeframe and up to a size of 10 MB, by using an ITZBund instance, too.

### Screenshots
not available (cannot reproduce this on our testing systems)

### System Details

* Device:
* SORMAS version: 1.59.1
* Android version/Browser: firefox
* Server URL: not available 
* User Role: all with access to _EDIT Case and contact rights

**Comment by Marko-ilmkreis** [comment]
@maxiheyner I will test it tomorrow morning after our update and upload screenshots if its failing.

**Comment by maxiheyner** [comment]
@Marko-ilmkreis thanks a lot!

**Comment by MateStrysewske** [comment]
Are there any error messages?

**Comment by maxiheyner** [comment]
@MateStrysewske yes the uploads starts and after a few minutes the user is told that all uploads were stopped, but we didn't receive any screenshots yet

**Comment by Marko-ilmkreis** [comment]
@MateStrysewske @maxiheyner there is no error-message, it is just not working. I tried is with the updated version and made a screen-recording.

https://user-images.githubusercontent.com/74791118/118547491-ad283d80-b759-11eb-883e-0baaec5ea357.mp4

**Comment by maxiheyner** [comment]
@Marko-ilmkreis many thanks for sending the video!

**Comment by MateStrysewske** [comment]
Maybe it's related to the server configuration and security restrictions there, since it's not reproducable on test systems? I'm a bit at a loss here. @syntakker Not sure whether you initially implemented the document upload - do you have any idea what could cause this?

**Comment by Marc-FS** [comment]
We have the same problem here. 
It seems that after clicking on some other part of the page after uploading, that the notification pops up. 

![image](https://user-images.githubusercontent.com/79253475/118620311-ac2cf580-b7c5-11eb-9df7-95cbadb6fe50.png)


- ITZ Bund Version 1.59.1
- Browser: Chrome and Firefox 
- file size: 900KB, .jpg and 800KB .pdf

**Comment by syntakker** [comment]
@MateStrysewske no, I implemented the document templates, but this is independent of document upload.
However it might be worth checking whether template upload is affected, too, since both write to the server's disk.

Beyond this:
- is the server log available?
- the path for document storage is configured in the `sormas.properties` file (property `documents.path`, default is `/opt/sormas/documents/`). This directory must exist and be writable by SORMAS.

**Comment by syntakker** [comment]
Please check also if the documents that have been uploaded before are still available. If not, the document upload directory might not be configured correctly.

**Comment by kwa20** [comment]
@syntakker @MateStrysewske Maybe this problem has something to do with proxy settings? This seems to be a recurring issue.

**Comment by Marko-ilmkreis** [comment]
@syntakker @MateStrysewske The previously uploaded templates are still there and can be used.

**Comment by syntakker** [comment]
@Marko-ilmkreis to avoid confusion: Do you mean the previously uploaded *documents* are still there, or the previously uplodaded *document templates* are still there?

**Comment by Marko-ilmkreis** [comment]
`the previously uplodaded document templates are still there`

This is right.

We were not able to upload documents.

**Comment by syntakker** [comment]
@MateStrysewske who can check on the instance whether the folder `/opt/sormas/documents/` exists and is writable by SORMAS?
If `documents.path` in `sormas.properties` is configured and points to some other folder, this should be checked for this folder.

**Comment by nliakm** [comment]
We investigated that problem further and it looks like, this is not a problem related with SORMAS itself but to the specific infrastructure the SORMAS hosts are running in.
It seems like the upload is blocked by the web application firewall. We contacted the administrators of that infrastructure to solve that problem.

**Comment by herrling-LRA-KYF** [comment]
Its also not working for events:

GET https://landratsamt-kyffhaeuserkreis.sormas.bund.de/error_path/400.html?al_req_id=YKO@h9@srQnTdfUXMWhL8QAAAl4 400 (Bad Request)

<html>
<head>
<base href="/error_path/">
<title>400 Bad Request</title>
</head>
<body>
<h1>Bad Request</h1>
<p>The server detected a syntax error in your request.<br>
Check your request and all parameters.</p>

**Comment by GA-LKGI** [comment]
> We have the same problem here.
> It seems that after clicking on some other part of the page after uploading, that the notification pops up.
> 
> ![image](https://user-images.githubusercontent.com/79253475/118620311-ac2cf580-b7c5-11eb-9df7-95cbadb6fe50.png)
> 
> * ITZ Bund Version 1.59.1
> * Browser: Chrome and Firefox
> * file size: 900KB, .jpg and 800KB .pdf

Exactly the same for us.

**Comment by Marko-ilmkreis** [comment]
@nliakm do we need to open a case at Netzlink or are you follow-up this things?

**Comment by nliakm** [comment]
@MateStrysewske  We got an answer from the administrators. This is nothing they will fix:

We can see for that specific request following log-entry:
```
2021-05-18 13:36:55 +02:00		 RProxyNDA01
RProxyNDA01 SG_child[243575]: {"log_id":"WR-SG-BLOCK-108-08","log_cat":"Y","req_id":"YKOm15x8cUKCpGbnbwLqrAAAA9w","sess_id":"2032b4f2e93864163cbdcc343f53d9d7","mapping":"sormas.bund.de","audit_token":"<n/a>","geoip_continent":"EU","geoip_country":"DE","geoip_location":"52.33850,10.54050","vhost":"sormas.bund.de","vhost_ip":"192.168.174.37","vhost_port":"443","vhost_proto":"https","client_ip":"79.231.31.125","sess_auth":"false","attack_type":"Filter evasion","block_type":"Multipart Parser","constraint":"Syntax","th_mode":"BLOCK","message":"multipart boundary definition not found in content-type header \"multipart/form-data\""}
2021-05-18 13:36:55 +02:00		 RProxyNDA01
RProxyNDA01 SG_child[243575]: {"log_id":"WR-SG-SUMMARY","log_cat":"X","req_id":"YKOm15x8cUKCpGbnbwLqrAAAA9w","sess_id":"2032b4f2e93864163cbdcc343f53d9d7","mapping":"sormas.bund.de","audit_token":"<n/a>","entry_url":"https://sormas.bund.de/sormas-ui/APP/UPLOAD/0/579/target/30c2265f-baf4-45e8-99fb-62873c5cdf74","vhost":"sormas.bund.de","vhost_ip":"192.168.174.37","vhost_port":"443","vhost_proto":"https","http_method":"POST","http_status":"400","entry_path":"/sormas-ui/APP/UPLOAD/0/579/target/30c2265f-baf4-45e8-99fb-62873c5cdf74","entry_query":"<n/a>","vhost_proto_vers":"1.1","sess_auth":"false","backend_url":"<n/a>","http_referrer":"https://test-essen.sormas.bund.de/sormas-ui/","req_size":"27776","time_total":"36928","time_filter":"75","client_ip":"79.231.31.125","front_src_ip":"79.231.31.125","front_src_port":"50556","geoip_continent":"EU","geoip_country":"DE","geoip_location":"52.33850,10.54050","action":"blocked","attack_type":"Filter evasion","block_type":"Multipart Parser","message":"Request processed"}
```
This means our WAF blocks this request, because the header multipart/form-data contains no boundary.
We recommend investigating the application regarding this topic.

Please see section 4.1 in https://www.ietf.org/rfc/rfc2388.txt

**Comment by MateStrysewske** [comment]
@vidi42 Is this something you can fix?

**Comment by vidi42** [comment]
@MateStrysewske I'll look into it.

**Comment by vidi42** [comment]
@nliakm can we find out if this issue is only happening for Documents upload or is it reproducible for imports from CSVs as well?

**Comment by vidi42** [comment]
Just test and it seems the issue is only related to multiple documents upload. We will provide a fix where Multiple Documents Upload can be enabled/disabled trough a feature configuration, as having it disabled seems to fix it.

This is until a better fix can be found.

**Comment by MateStrysewske** [comment]
@kwa20 The component is still a multi-upload component, even if you only upload a single file through it. According to the tests by @vidi42, file upload works in other places where we don't use a multi-upload component (e.g. uploading import files)

**Comment by kwa20** [comment]
@MateStrysewske I see, thanks for the clarification!

**Comment by vidi42** [comment]
@GA-LKGI I tried to reproduce the issue with Document Template upload and that seemed to work on the same environment where Multi Document Upload was failing. In which part of the application was it failing for Single Document Upload?

**Comment by GA-LKGI** [comment]
@vidi42 The upload for the document templates works fine. Just for documents on the other hand, the issue appears after choosing the file we want to upload. After selecting the file nothing happens as long as we click somewhere on the page. Then "all uploads have been interrupted" is shown. Like said, also for single document upload.

**Comment by RazvanHodisan** [comment]
I think the issue described in the above comment is an edge case. If you select one or more documents to upload and you wait for the pop up to complete then the documents will be uploaded. However, if you click on upload then instantly click anywhere else before the pop up is displayed the upload will be interrupted with the respective error message. Might be a good idea to add this edge case into a different issue. 

Validated on Version: 1.61.0-SNAPSHOT

**Comment by Marko-ilmkreis** [comment]
@RazvanHodisan No, even after few minutes of waiting for a file less then 400 kB there was no sucess.

**Comment by vidi42** [comment]
> I think the issue described in the above comment is an edge case. If you select one or more documents to upload and you wait for the pop up to complete then the documents will be uploaded. However, if you click on upload then instantly click anywhere else before the pop up is displayed the upload will be interrupted with the respective error message. Might be a good idea to add this edge case into a different issue.
> 
> Validated on Version: 1.61.0-SNAPSHOT

The issue @RazvanHodisan describes and the one in this comment https://github.com/hzi-braunschweig/SORMAS-Project/issues/5452#issuecomment-843188962 are definitely looking similar, however they are not. 

The issue described initially is the one addressed by the fix described here https://github.com/hzi-braunschweig/SORMAS-Project/issues/5452#issuecomment-844948549.

However @RazvanHodisan's comment points to an edge case, where if the documents upload started, if you click outside the popup or click before the popup is shown, the system interrupts the download automatically.

**Comment by Candice-Louw** [comment]
Can confirm that this is happening on CH servers as well.  On stable-release-ch (test server) it works fine, but on test.sormas.ch it does not upload at all (403 forbidden error).  CH server config (WAF) possible blocker as pointed out by @nliakm .
![image](https://user-images.githubusercontent.com/65529128/119635697-3bfd2000-be14-11eb-976e-ed05c701a921.png)

**Comment by kwa20** [comment]
@markusmann-vg @MateStrysewske @StefanKock 

https://github.com/hzi-braunschweig/SORMAS-Project/issues/5452#issuecomment-843868773

How do we address the fix for the actual multi-component problem aside from the single upload quick fix? Do we open another issue for it? If so, can someone specify this who knows what needs to be done?

**Comment by herrling-LRA-KYF** [comment]
What is the status here?
For me it is a little bit confusing because the ticket is closed and it is not working at the moment.

**Comment by vidi42** [comment]
@herrling-LRA-KYF what exactly is not working at the moment?
This ticket was about introducing a feature configuration which can enable or disable the multi document upload feature.
By disabling it people should now be able to upload documents to cases, contacts and events one at a time (single document upload).

**Comment by herrling-LRA-KYF** [comment]
I cant upload a single document. How can I disable the multi document upload feature?
If I try to upload one document I get this error: https://github.com/hzi-braunschweig/SORMAS-Project/issues/5452#issuecomment-843188962

**Comment by vidi42** [comment]
Starting with [SORMAS 1.60.0](https://github.com/hzi-braunschweig/SORMAS-Project/releases/tag/v1.60.0) there is a new feature configuration `DOCUMENTS_MULTI_UPLOAD` which can be disabled directly in the DB (table `featureconfiguration`).
